sdk/src/entrypoint_native.rs

```diff
@@ -265,3 +272,33 @@ pub trait Logger {
     /// Log a message
     fn log(&mut self, message: &str);
 }
+
+/// Program executor
+pub trait Executor: Send + Sync {
+    /// Execute the program
+    fn execute(
+        &self,
+        program_id: &Pubkey,
+        keyed_accounts: &[KeyedAccount],
+        instruction_data: &[u8],
+        invoke_context: &mut dyn InvokeContext,
+    ) -> Result<(), InstructionError>;
+}
+#[cfg(RUSTC_WITH_SPECIALIZATION)]
+impl AbiExample for Arc<dyn Executor> {
```

Comment:
@ryoqun Is this the preferred way to provide an example of `Arc<dyn Executor>`?

Reply:
Yeah, this looks good!